### PR TITLE
adding `ifft2` method to ops

### LINKED ISSUE #20437: Add `ifft2` method to ops
I'm curious why there is no `ops.ifft2`.   Given that there is already `fft` and `fft2`,  implementing one is trivial.

Here is an example of what an `ifft2` would look like:

```python
import keras
def keras_ops_ifft2(fft_real,fft_imag):
    """
    Inputs are real and imaginary parts of array
    of shape [...,H,W],[...,H,W] , where the last two 
    dimensions correspond tot he image dimensions

    Returns tuple containing the real and imaginary parts
    of the ifft2 

    Test:
       from keras import ops
       X = np.random.rand( 1,1,11,11 ).astype(np.float32)
       X_real,X_imag = ops.real(X), ops.imag(X)
       X_fft_real,X_fft_imag = keras.ops.fft2((X_real,X_imag))
       X_recon,_ = keras_ops_ifft2(X_fft_real,X_fft_imag)
       np.allclose(X,X_recon,atol=1e-6)
       
    """
    H = ops.cast(ops.shape(fft_real)[-2],'float32') # height
    W = ops.cast(ops.shape(fft_real)[-1],'float32') # width

    # Conjugate the input
    real_conj, imag_conj = fft_real, -fft_imag

    # Compute FFT of conjugate
    fft = ops.fft2((real_conj, imag_conj))

    return fft[0] / (H*W), -fft[1] / (H*W)
```

### PR DESCRIPTION
**Pull request by rohithpudari**
Added implementation of `ifft2` to ops.

fixes #20437

### CODE REVIEW

File: keras/src/backend/jax/math.py [hunk 1]
```diff
@@ -123,6 +123,15 @@ def fft2(x):
     return jnp.real(complex_output), jnp.imag(complex_output)
 
 
+def ifft2(x):
```
Comment (fchollet) [inline keras/src/backend/jax/math.py:126]: Please use the built-in version https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.fft.ifft2.html

File: keras/src/backend/tensorflow/math.py [hunk 2]
```diff
@@ -113,6 +113,15 @@ def fft2(x):
     return tf.math.real(complex_output), tf.math.imag(complex_output)
 
 
+def ifft2(x):
+    real, imag = x
+    H = cast(tf.shape(real)[-2], "float32")
```
Comment (fchollet) [inline keras/src/backend/tensorflow/math.py:118]: Please use lowercase variable names.

File: keras/src/backend/numpy/math.py [hunk 3]
```diff
@@ -144,6 +144,15 @@ def fft2(x):
     return np.array(real), np.array(imag)
 
 
+def ifft2(x):
```
Comment (fchollet) [inline keras/src/backend/numpy/math.py:147]: Please use the built-in version. https://numpy.org/doc/stable/reference/generated/numpy.fft.ifft2.html

File: keras/src/backend/torch/math.py [hunk 4]
```diff
@@ -203,6 +203,15 @@ def fft2(x):
     return complex_output.real, complex_output.imag
 
 
+def ifft2(x):
```
Comment (fchollet) [inline keras/src/backend/torch/math.py:206]: https://pytorch.org/docs/stable/generated/torch.fft.ifft2.html

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/src/backend/jax/math.py (hunk 1 above)
Comment (rohithpudari) [reply keras/src/backend/jax/math.py:126]: updated to use built-in version

File: keras/src/backend/numpy/math.py (hunk 3 above)
Comment (rohithpudari) [reply keras/src/backend/numpy/math.py:147]: updated to use built-in version

File: keras/src/backend/torch/math.py (hunk 4 above)
Comment (rohithpudari) [reply keras/src/backend/torch/math.py:206]: updated to use built-in version

File: keras/src/backend/tensorflow/math.py (hunk 2 above)
Comment (rohithpudari) [reply keras/src/backend/tensorflow/math.py]: updated to use lowercase names

**Review by fchollet — APPROVED** [review]
LGTM -- thank you for the contribution!

**Pull request merged**